### PR TITLE
FIX: check if contour level in format dictionary, or return a default

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

As per #9742 the user could pass a format dictionary to `clabel` that doesn't contain a contour level.  Normally this would be a too-bad-so-sad sort of thing, except if there are not contours we *change* the value of the contour ( 6ac197f ).

This fix is maybe suboptimal in that the user may wonder why a contour label sneaks in with a different format.  However, I doubt too many users would specify the contour formats but not the levels, so this really only handles the edgecase in #9742.

## PR Checklist

- [ ] Has Pytest style unit tests
- [x] Code is PEP 8 compliant

### CODE REVIEW

File: lib/matplotlib/contour.py [hunk 1]
```diff
@@ -328,7 +328,10 @@ def get_text(self, lev, fmt):
             return lev
         else:
             if isinstance(fmt, dict):
-                return fmt[lev]
```
Comment (tacaswell) [inline lib/matplotlib/contour.py]: Doing this as `fmt.get(lev, '%1.3f')` is a bit terser.

File: lib/matplotlib/tests/test_contour.py [hunk 2]
```diff
@@ -143,6 +143,25 @@ def test_contour_empty_levels():
...
+    # https://github.com/matplotlib/matplotlib/pull/9743
+    x = np.arange(9)
+    z = np.zeros((9, 9))
+
+    fig, ax = plt.subplots()
+    fmt = {1.: '%1.2f'}
+    with pytest.warns(UserWarning) as record:
+        cs = ax.contour(x, x, z, levels=[1.])
+        ax.clabel(cs, fmt=fmt)
+    assert len(record) == 1
```
Comment (efiring) [inline lib/matplotlib/tests/test_contour.py]: Shouldn't this be inside the `with` block?

Comment (jklymak) [reply lib/matplotlib/tests/test_contour.py]: I just copied the previous test.   Seems to work but happy to change if wrong

Comment (efiring) [reply lib/matplotlib/tests/test_contour.py]: OK, I see now where this style is shown in the pytest documentation, so no problem.

**Pull request merged**